sawmill_api/handlers/settings.py

```diff
@@ -0,0 +1,57 @@
+from sawmill_api import utils, settings
+from sawmill_api.lib import oltp
+
+import flask
+import pydantic
+
+
+settings_api = flask.Blueprint("settings_api", __name__, url_prefix="/api/1/settings")
+log = utils.get_logger(__name__)
+
+# A bit long, but enables us to just add settings to module
+# and auto-magically reflect them in the API. Yay automation!
+SettingsAPI = pydantic.create_model(
+    "SettingsAPI",
+    **{
+        settings.title_to_dash(section_name): settings_section.__class__
+        for section_name, settings_section in list(settings)
+    },
+)
+
+
+@settings_api.route("/", methods=["GET"])
+def get_settings():
+    """Read the current settings being used by Sawmill."""
+    data = {}
+    for section_name, current_settings in list(settings):
+        data[settings.title_to_dash(section_name)] = current_settings
+    return SettingsAPI(**data).model_dump_json(), 200
+
+
+@settings_api.route("/<section>", methods=["PUT"])
+def update_setting(section):
```

Comment:
The _keen eyed_ reviews will notice that Sawmill is storing _all_ section settings in the DB instead of just the updated ones. I'll likely regret this giving how fate seems to work out. For now, I can't think of a horrible reason to not do this.